pallets/payment/src/lib.rs

```diff
@@ -233,7 +240,16 @@ pub mod pallet {
 					T::Asset::unreserve(payment.asset, &from, payment.incentive_amount);
 					// unreserve the amount to the recipent
 					T::Asset::unreserve(payment.asset, &to, payment.amount);
-
+					// calculate fee charged for transaction
```

Comment:
I think it's better for the fee to be calculated at the moment of creation of the payment and not when released to avoid "surprises". As a customer when you pay for an article you want to know well from the beginning what you are paying, if we do this in the release stage is hard to predict what the fee will be. We can calculate at the moment of creation and store the amount to be charged in the payment detail in a `fees` field, then we transfer on release. Probably also wise to reserve the amount to make sure we can pay the marketplace(what happens if the account has no funds from the moment the payment is created and released? is payment cancelled?), named reserves will probably help making that bookkeeping clearer 🤔